openvisualizer/moteProbe/moteProbe.py

```diff
@@ -150,6 +99,59 @@ def findSerialPorts(isIotMotes=False, isTestbedMotes=False):
 
 #============================ class ===========================================
 
+class findOpenTestbedMotes(object):
+
+    OPENTESTBED_RESP_STATUS_TIMEOUT     = 10
+
+    def __init__(self):
+    
+        # motelist 
+        self.opentestbed_motelist = []
+        
+        # create mqtt client
+        self.mqtt_client                = mqtt.Client('FindMotes')
+        self.mqtt_client.on_connect     = self._on_mqtt_connect
+        self.mqtt_client.on_message     = self._on_mqtt_message
+        self.mqtt_client.connect(OPENTESTBED_BROKER_ADDRESS)
+        self.mqtt_client.loop_start()
+        
+        # wait for a while to gather the response from otboxes
+        time.sleep(self.OPENTESTBED_RESP_STATUS_TIMEOUT)
+        
+        # close the client and return the motes list
+        self.mqtt_client.loop_stop()
+        
+        print "{0} motes are found".format(len(self.opentestbed_motelist))
+        
+    def get_opentestbed_motelist(self):
+        
+        return self.opentestbed_motelist
```

Comment:
race condition. How do you ensure you call `get_opentestbed_motelist()` after the motes have been discovered?

Reply:
@twatteyne In the _init_ function, it already wait for 10 seconds. That will make sure the messages from all otboxes are received.